src/details-page.js

```diff
@@ -0,0 +1,165 @@
+const loader = document.getElementsByClassName('loader')[0];
+const detailsSlot = document.getElementById('details');
+
+// Extract CCA3 country code from current window's url params and remove loader
+const searchStr = location.search.substring(1);
+let [param, cca3] = searchStr.split('=', 2);
+const loadDetails = async () => {
+    const country = await getCountryByCCA3(cca3);
+    detailsSlot.append( await convertToDetailPage(country));
+}
+loadDetails()
+.then( () => { loader.classList.add('hide') })
+
+// Global Data Not Found object
+const DNF = '<strong class=\'warning\'>none</strong>';
+
+const convertToDetailPage = async jsonObj => {
+    const flagSrc = jsonObj.flags.svg || jsonObj.flags.png || DNF; // prefer SVG over PNG
+    const flagAltTxt = jsonObj.flags.alt || '';
+    const countryName = jsonObj.name.official || jsonObj.name.common || DNF;
```

Comment:
Your comment say _prefer SVG over PNG_ but the next line saved a flag alt text in a variable. 
Confusing